apps/wallet-mobile/src/WalletNavigator.tsx

```diff
@@ -174,26 +196,30 @@ export const WalletNavigator = () => {
   }
 
   return (
-    <Stack.Navigator
-      screenOptions={{
-        headerShown: false /* used only for transition */,
-        detachPreviousScreen: false /* https://github.com/react-navigation/react-navigation/issues/9883 */,
-      }}
-    >
-      <Stack.Screen name="wallet-selection" component={WalletSelectionScreen} />
+    <BrowserProvider>
+      <Stack.Navigator
+        screenOptions={{
+          headerShown: false /* used only for transition */,
+          detachPreviousScreen: false /* https://github.com/react-navigation/react-navigation/issues/9883 */,
+        }}
+      >
+        <Stack.Screen name="wallet-selection" component={WalletSelectionScreen} />
 
-      <Stack.Screen name="main-wallet-routes" component={WalletTabNavigator} />
+        <Stack.Screen name="main-wallet-routes" component={WalletTabNavigator} />
 
-      <Stack.Screen name="nft-details-routes" component={NftDetailsNavigator} />
+        <Stack.Screen name="nft-details-routes" component={NftDetailsNavigator} />
 
-      <Stack.Screen name="settings" component={SettingsScreenNavigator} />
+        <Stack.Screen name="settings" component={SettingsScreenNavigator} />
 
-      <Stack.Screen name="voting-registration" component={VotingRegistration} />
+        <Stack.Screen name="voting-registration" component={VotingRegistration} />
 
-      <Stack.Screen name="toggle-analytics-settings" component={ToggleAnalyticsSettingsNavigator} />
+        <Stack.Screen name="toggle-analytics-settings" component={ToggleAnalyticsSettingsNavigator} />
 
-      <Stack.Screen name="governance" component={GovernanceNavigator} />
-    </Stack.Navigator>
+        <Stack.Screen name="governance" component={GovernanceNavigator} />
+
+        <Stack.Screen name="browser" component={BrowserNavigator} />
```

Comment:
I think cannot move BrowserProvider inside DiscoverNavigator because DiscoverNavigator is a bottom tab screen, cannot show full screen when you're going to Browser View